Mage/src/main/java/mage/watchers/Watcher.java

```diff
@@ -75,18 +77,61 @@ public void reset() {
         condition = false;
     }
 
-    protected String getBasicKey(){
+    protected String getBasicKey() {
         return getClass().getSimpleName();
     }
 
     public abstract void watch(GameEvent event, Game game);
 
-    public <T extends Watcher> T copy(){
+    public <T extends Watcher> T copy() {
         try {
-            Constructor<? extends Watcher> constructor = this.getClass().getDeclaredConstructor(getClass());
+            List<?> constructors = Arrays.asList(this.getClass().getConstructors());
+            if (constructors.size() > 1) {
+                logger.error(getClass().getSimpleName() + " has multiple constructors");
+                return null;
+            }
+
+            Constructor<? extends Watcher> constructor = (Constructor<? extends Watcher>) constructors.get(0);
+            if (constructor.getParameterCount() > 0) {
+                logger.error(getClass().getSimpleName() + " constructor has arguments, should be 0 and inject the parameters by setters");
+                return null;
+            }
             constructor.setAccessible(true);
-            return (T) constructor.newInstance(this);
-        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+            T watcher = (T) constructor.newInstance();
+            List<Field> allFields = new ArrayList<>();
+            allFields.addAll(Arrays.asList(getClass().getDeclaredFields()));
+            allFields.addAll(Arrays.asList(getClass().getSuperclass().getDeclaredFields()));
+            for (Field field : allFields) {
+                field.setAccessible(true);
+                if (field.getType().isPrimitive()) {
+                    field.set(watcher, field.get(this));
+                }
+                else if(field.getType() == Step.class){
+                    field.set(watcher, field.get(this));
+                }
+                else if (field.getType() == Mana.class) {
+                    field.set(watcher, field.get(this));
+                } else if (field.getType() == UUID.class) {
```

Comment:
The code snippets you reference belong to Effects and Abilities. Those are different because they do reference hard copy() methods of their attributes. Watchers do not do that (except for 1 rare occassion)